client/my-sites/upgrades/index.js

```diff
@@ -289,5 +290,14 @@ module.exports = function() {
 			controller.siteSelection,
 			upgradesController.checkout
 		);
+
+		page(
+			'/start-trial/:domain/:plan_name?',
```

Comment:
We tend to favour having the domain at the end of the URL unless there is a compelling reason not to...